File: docs/samples/lua/jwt_encode.lua
-- Usage: lua jwt_encode.lua path/to/json/file.json path/to/private_key/pem

-- luarocks install luacrypto
-- luarocks install jwt
local cjson  = require 'cjson'
local crypto = require 'crypto'
local jwt    = require 'jwt'

-- read whole file into a string
function read_file(filepath)
    local f = io.open(filepath, "rb")
    local content = f:read("*all")
    f:close()
    return content
end

-- encode JWT passing the data table and the key path
function jwt_encode(data, key_path)
  local pkey = crypto.pkey.from_pem(read_file(key_path), true)
  local token, _ = jwt.encode(data, {
    alg = "RS256",
    keys = { private = pkey }
  })
  return token
end

-- get json data from file
local json_content = read_file(arg[1])
local data = cjson.decode(json_content)

-- get private key path
local key_path = arg[2]

-- encode our JWT token
local encoded_token = jwt_encode(data, key_path)

print(encoded_token)

File: docs/samples/python/jwt_encode.py
import sys
import json
import jwt
from jwt.contrib.algorithms.pycrypto import RSAAlgorithm

jwt.register_algorithm('RS256', RSAAlgorithm(RSAAlgorithm.SHA256))

def read_file(path):
  with open(path, 'r') as f:
    content = f.read()
  return content

def jwt_encode(data, key_path):
  pkey = read_file(key_path)
  return jwt.encode(data, pkey, algorithm='RS256')

data = json.loads(read_file(sys.argv[1]))
key_path = sys.argv[2]

print(jwt_encode(data, key_path))
